server/src/main/java/org/elasticsearch/cluster/desirednodes/DesiredNodesSettingsValidator.java

```diff
@@ -0,0 +1,124 @@
+                format(Locale.ROOT, "Nodes in positions [%s] contain invalid settings", nodeIndicesWithFailures)
+            );
+            for (Tuple<Integer, RuntimeException> exceptionTuple : exceptions) {
+                invalidSettingsException.addSuppressed(exceptionTuple.v2());
+            }
+            throw invalidSettingsException;
+        }
+    }
+
+    private void validate(DesiredNode node) {
+        if (node.version().before(Version.CURRENT)) {
+            throw new IllegalArgumentException(
+                format(Locale.ROOT, "Illegal node version [%s]. Only [%s] or newer versions are supported", node.version(), Version.CURRENT)
+            );
+        }
+
+        if (node.externalId() == null) {
+            throw new IllegalArgumentException(
+                format(Locale.ROOT, "[%s] or [%s] is missing or empty", NODE_NAME_SETTING.getKey(), NODE_EXTERNAL_ID_SETTING.getKey())
+            );
+        }
+
+        final Settings settings = node.settings();
+
+        final Set<Setting<?>> settingSet = new HashSet<>();
+        final List<String> unknownSettings = new ArrayList<>();
+        // If we're dealing with a node in a newer version it's possible
+        // that this node doesn't know about these. In that case we do a best
+        // effort and try to validate the settings that are known to this version.
+        final Settings.Builder updatedSettingsBuilder = Settings.builder().put(settings);
+        boolean hasOverriddenSettings = false;
+        for (String settingName : settings.keySet()) {
+            Setting<?> setting = clusterSettings.get(settingName);
+            if (setting == null) {
+                unknownSettings.add(settingName);
+                updatedSettingsBuilder.remove(settingName);
+            } else {
+                // Some settings might be defined using environmental information
+                // such as the number of available processors to define valid setting
+                // value ranges, in that case we should define new settings that take
+                // into account the node properties that we're evaluating.
+                Setting<?> maybeOverrideSetting = maybeOverride(node, setting);
+                hasOverriddenSettings |= (setting != maybeOverrideSetting);
+                settingSet.add(maybeOverrideSetting);
+            }
+        }
+
+        if (unknownSettings.isEmpty() == false && node.version().onOrBefore(Version.CURRENT)) {
+            throw new IllegalArgumentException(
+                format(Locale.ROOT, "Node [%s] has unknown settings %s", node.externalId(), unknownSettings)
+            );
+        }
+
+        final Settings updatedSettings = unknownSettings.isEmpty() ? settings : updatedSettingsBuilder.build();
+        final ClusterSettings updatedClusterSettings = hasOverriddenSettings
+            ? new ClusterSettings(updatedSettings, settingSet)
```

Comment:
I think we cannot really assume that this is safe. A couple of problems:

1. Assume an enum and the newer version added a new option. Using that would fail on current version. Similarly for any setting that has expanded set of legal values.
2. Assume a new setting is added that serves as the default for another setting and the orchestrator uses the new setting on a newer version rather than the original. This could in principle ruin validation.

We could verify the settings still and produce some sort of warning (I assume it will still be rare). But for now I would prefer to simply skip validation when the version is newer than current.